packages/grid/x-data-grid/src/components/cell/GridEditSingleSelectCell.tsx

```diff
@@ -12,18 +12,39 @@ import { useGridRootProps } from '../../hooks/utils/useGridRootProps';
 import { GridEditModes } from '../../models/gridEditRowModel';
 import { GridEvents } from '../../models/events/gridEvents';
 import { GridColDef, ValueOptions } from '../../models/colDef/gridColDef';
+import { getValueFromValueOptions } from './editCellUtils';
 
-const renderSingleSelectOptions = (option: ValueOptions) =>
-  typeof option === 'object' ? (
-    <MenuItem key={option.value} value={option.value}>
-      {option.label}
-    </MenuItem>
+interface SelectOptionProps {
+  isNative: boolean;
+  children: React.ReactNode;
+  value: any;
+}
```

Comment:
This informs typescript that `...props` can contain any porps of the `MenuItem`

```suggestion
interface SelectOptionProps extends MenuItemProps {
  isNative: boolean;
  children: React.ReactNode;
  value: any;
}
```

The type `MenuItemProps` can be imported from `'@mui/material/MenuItem'`

Reply:
I considered extending `MenuItemProps`, but `props` from `MenuItemProps` cannot be passed to `option` tag.
What would you think about seperating `optionProps` and `menuItemProps` to pass accurate `props`
It will be like...
```ts
type SelectOptionProps = {
  isNative: boolean;
  children: React.ReactNode;
  value: any;
  optionProps?: React.DetailedHTMLProps<
    React.OptionHTMLAttributes<HTMLOptionElement>,
    HTMLOptionElement
  >;
  menuItemProps?: MenuItemProps;
};

function SelectOption({
  isNative,
  children,
  value,
  optionProps,
  menuItemProps,
  ...props
}: SelectOptionProps) {
  return isNative ? (
    <option value={value} {...optionProps} {...props}>
      {children}
    </option>
  ) : (
    <MenuItem value={value} {...menuItemProps} {...props}>
      {children}
    </MenuItem>
  );
}
```